File: app/src/main/java/com/shentu/wallpaper/model/entity/Subject.java
package com.shentu.wallpaper.model.entity;

import com.chad.library.adapter.base.entity.MultiItemEntity;

public class Subject implements MultiItemEntity {
    public static final int ITEM_VIEW_1 = 1;//魅族壁纸
    public static final int ITEM_VIEW_2 = 2;//cosplay
    public static final int ITEM_VIEW_3 = 3;//性感美女

    public int id;
    public String name;
    public String cover;
    public String description;
    public String tag;
    public String created;
    public int bgColor;
    public String cover_1;
    public String cover_2;
    public int type;

    public Subject() {
        type = ITEM_VIEW_2;
    }

    @Override
    public int getItemType() {
        return type;
    }
}

File: app/src/main/java/com/shentu/wallpaper/model/entity/Subject.kt
package com.shentu.wallpaper.model.entity

import com.chad.library.adapter.base.entity.MultiItemEntity

class Subject : MultiItemEntity {
    var id = 0
    var name: String=""
    var cover: String = ""
    var description: String = ""
    var tag: String? = null
    var created: String=""
    var bgColor = 0
    var cover_1: String = ""
    var cover_2: String = ""
    var type: Int
    override fun getItemType(): Int {
        return type
    }

    companion object {
        const val ITEM_VIEW_1 = 1 //魅族壁纸
        const val ITEM_VIEW_2 = 2 //cosplay
        const val ITEM_VIEW_3 = 3 //性感美女
    }

    init {
        type = ITEM_VIEW_2
    }
}